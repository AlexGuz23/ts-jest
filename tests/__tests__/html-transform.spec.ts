import * as tsJest from '../../dist';
import mockJestConfig from '../__helpers__/mock-jest-config';

const TEST_CASE = 'simple';
const FILENAME = 'some-file.html';

const fileContent = `<div class="html-test">
  <span class="html-test__element">This is element</span>
  <code>This is a backtilt \`</code>
</div>`;

describe('Html transforms', () => {
  it('transforms html if config.globals.__TRANSFORM_HTML__ is set', () => {
    let jestConfig;

    // get the untransformed version
    jestConfig = mockJestConfig(TEST_CASE);
    const untransformed = tsJest.process(
      fileContent,
      `${jestConfig.rootDir}/${FILENAME}`,
      jestConfig,
    );
    expect(untransformed).toBe(fileContent);
    expect(untransformed).toMatchSnapshot('untransformed');

    // ... then the one which should be transformed
    jestConfig = {
      ...mockJestConfig(TEST_CASE),
      globals: { __TRANSFORM_HTML__: true },
    };
    const transformed = tsJest.process(
      fileContent,
      `${jestConfig.rootDir}/${FILENAME}`,
      jestConfig,
    ) as string;
    expect(transformed).not.toBe(fileContent);
    expect(transformed).toMatchSnapshot('source');

    // ... finally the result of a `require('module-with-transformed-version')`
    const value = eval(
      `(function(){const module={}; ${transformed}; return module.exports;})()`,
    );
    expect(value).toMatchSnapshot('module');
    // the value should be the same string as the source version
    expect(value).toBe(fileContent);
  });
});
