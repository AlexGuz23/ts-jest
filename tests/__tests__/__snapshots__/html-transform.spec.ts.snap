// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Html transforms transforms html if config.globals.__TRANSFORM_HTML__ is set: module 1`] = `
"<div class=\\"html-test\\">
  <span class=\\"html-test__element\\">This is element</span>
  <code>This is a backtilt \`</code>
</div>"
`;

exports[`Html transforms transforms html if config.globals.__TRANSFORM_HTML__ is set: source 1`] = `"module.exports=\\"<div class=\\\\\\"html-test\\\\\\">\\\\n  <span class=\\\\\\"html-test__element\\\\\\">This is element</span>\\\\n  <code>This is a backtilt \`</code>\\\\n</div>\\";"`;

exports[`Html transforms transforms html if config.globals.__TRANSFORM_HTML__ is set: untransformed 1`] = `
"<div class=\\"html-test\\">
  <span class=\\"html-test__element\\">This is element</span>
  <code>This is a backtilt \`</code>
</div>"
`;
